File: src/adder.rs
use log::info;

#[derive(Default)]
pub struct Adder {
    count: i64,
}

impl Adder {
    pub fn add(&mut self, value: i64) {
        info!("Adder::add()");
        self.count += value;
    }

    pub fn tell(&self) -> i64 {
        info!("Adder::tell()");
        self.count
    }
}

File: test/Adder.cpp
#include <iostream>

class Adder
{
    void *internal;

public:
    Adder();

    ~Adder();

    void add(int64_t value);

    int64_t tell() const;
};

Adder::Adder()
{
}
Adder::~Adder()
{
}

void Adder::add(int64_t value)
{
}

int64_t Adder::tell() const
{
    return 0;
}